pkg/tests/end_to_end_tests/metric_ingest_bench_test.go

```diff
@@ -74,3 +77,30 @@ func BenchmarkMetricIngest(b *testing.B) {
 		b.StopTimer()
 	})
 }
+
+func BenchmarkNewSeriesIngestion(b *testing.B) {
+	seriesGen, err := testsupport.NewSeriesGenerator(100, 100, 4)
+	require.NoError(b, err)
+
+	batches := seriesGen.GetBatches(2000)
+
+	withDB(b, "bench_e2e_new_series_ingest", func(db *pgxpool.Pool, t testing.TB) {
+		metricsIngestor, err := ingestor.NewPgxIngestorForTests(pgxconn.NewPgxConn(db), &ingestor.Cfg{
+			NumCopiers:              8,
+			InvertedLabelsCacheSize: cache.DefaultConfig.InvertedLabelsCacheSize,
+		})
+		require.NoError(b, err)
+		defer metricsIngestor.Close()
+
+		b.Run("New Series Ingestion", func(b *testing.B) {
```

Comment:
Looks like benchmark iteration count (`b.N`) is ignored. We should honer that right?

Reply:
Since we are writing to DB benchmark iterations would result in duplicates. Ingesting fairly large amount of data should help getting more stable results but obviously it's not ideal. Do you maybe have some ideas?

Reply:
we anyways run the benchmark 10 time due to `-count` before calculating the difference with the master branch. Hence, we don't have to care much about `b.N` here. That said, I think `b.Run()` takes care internally of the benchmark iteration.

Reply:
In that case, can we consider generating number of series based on b.N?

Reply:
That is a good point, but N is a non-static value, right? So during the benchmarks, the value of N will change, meaning generating series on runtime might not be a good thing.

Reply:
> Since we are writing to DB benchmark iterations would result in duplicates.

How about using different DB instance for each iteration to avoid duplication? Also this would reproduce the same scenario on each iteration.

The current approach is,
```
1. Create DB
2. Ingest batch 1,
3. ....
4. Ingest batch n
```
Each iteration being adds more data to the existing db which is not deterministic.

Recommended approach is,
```
for i :=0; i < b.N; i++ {
1. Create DB
2. Ingest batch 1,
3. ....
4. Ingest batch n
}
```

Reply:
I misinterpreted that `b.N` is based on `-count` flag. Ignore my previous comment.

Reply:
I'm bit confused here, When I run this benchmark, I see the following report.

```
$ go test -test.bench NewSeriesIngestion -run=^$ -cpu=1 -timeout=6h -count=10 ./pkg/tests/end_to_end_tests

BenchmarkNewSeriesIngestion/New_Series_Ingestion                       1        17720620125 ns/op    62112104 B/op     520193 allocs/op
BenchmarkNewSeriesIngestion/New_Series_Ingestion                1000000000               0.0001638 ns/op            0 B/op          0 allocs/op
BenchmarkNewSeriesIngestion/New_Series_Ingestion                1000000000               0.0001365 ns/op            0 B/op          0 allocs/op
BenchmarkNewSeriesIngestion/New_Series_Ingestion                level=info ts=2022-07-26T11:28:29.708Z caller=throughput.go:76 msg="ingestor throughput" samples/sec=2000 metrics-max-sent-ts=1970-01-01T00:00:00.001Z
1000000000               0.0001356 ns/op               0 B/op          0 allocs/op
BenchmarkNewSeriesIngestion/New_Series_Ingestion                1000000000               0.0001402 ns/op            0 B/op          0 allocs/op
BenchmarkNewSeriesIngestion/New_Series_Ingestion                1000000000               0.0001354 ns/op            0 B/op          0 allocs/op
BenchmarkNewSeriesIngestion/New_Series_Ingestion                1000000000               0.0001322 ns/op            0 B/op          0 allocs/op
BenchmarkNewSeriesIngestion/New_Series_Ingestion                1000000000               0.0001254 ns/op            0 B/op          0 allocs/op
BenchmarkNewSeriesIngestion/New_Series_Ingestion                1000000000               0.0001434 ns/op            0 B/op          0 allocs/op
BenchmarkNewSeriesIngestion/New_Series_Ingestion                1000000000               0.0001302 ns/op            0 B/op          0 allocs/op
```
IIUC, only first time it ran successfully and the rest of the 9 times it gives number of iteration as 1000000000. Am I missing something here?

Reply:
> How about using different DB instance for each iteration to avoid duplication?

We could try that. That benchmark might run for quite a long time since in each iteration we need to insert a lots of data.

Reply:
Interesting. I wonder if that's because of test caching

Reply:
I'm yet to validate, but I think withDB fails and returns silently on the second iteration.

Reply:
@niksajakovljevic Looks like you are right. When running a benchmark inside `b.Run`, it seem to be catching the result because on each iteration benchmark name is same(first arg). Removing b.Run fixes the issue.

Reply:
b.Run is now removed :)